src/csharp/Microsoft.Spark/Broadcast.cs

```diff
@@ -0,0 +1,265 @@
+        /// <param name="value">Broadcast value of type object</param>
+        /// <returns></returns>
+        private JvmObjectReference CreateBroadcast_V2_3_2_Above(
+            SparkContext sc,
+            string tempDir,
+            JvmObjectReference sparkContext,
+            object value)
+        {
+            var pythonBroadcast = (JvmObjectReference)sparkContext.Jvm.CallStaticJavaMethod(
+                    "org.apache.spark.api.python.PythonRDD",
+                    "setupBroadcast",
+                    _path);
+            SparkConf sparkConf = sc.GetConf();
+            var encryptionEnabled = bool.Parse(
+                sparkConf.Get("spark.io.encryption.enabled",
+                "false"));
+
+            if (encryptionEnabled)
+            {
+                throw new NotImplementedException(
+                    "Broadcast encryption is not supported yet.");
+            }
+            else
+            {
+                WriteBroadcastValueToFile(tempDir, value);
+            }
+
+            return (JvmObjectReference)sparkContext.Jvm.CallStaticJavaMethod(
+                "org.apache.spark.sql.api.dotnet.SQLUtils",
+                "createBroadcast",
+                sparkContext,
+                pythonBroadcast);
+        }
+    }
+
+    /// <summary>
+    /// Global registry to store the object value of all active broadcast variables from
+    /// the workers. This registry is only used on the worker side when Broadcast.Value() is called
+    /// through a UDF.
+    /// </summary>
+    internal static class BroadcastRegistry
+    {
+        private static ConcurrentDictionary<long, object> s_registry = 
+            new ConcurrentDictionary<long, object>();
+
+        /// <summary>
+        /// Function to add the value of the broadcast variable to s_registry.
+        /// </summary>
+        /// <param name="bid">Id of the Broadcast variable object to add</param>
+        /// <param name="value">Value of the Broadcast variable</param>
+        public static void Add(long bid, object value)
+        {
+            s_registry.TryAdd(bid, value);
+        }
+
+        /// <summary>
+        /// Function to remove the Broadcast variable from s_regitry.
+        /// </summary>
+        /// <param name="bid">Id of the Broadcast variable object to remove</param>
+        public static void Remove(long bid)
+        {
+            s_registry.TryRemove(bid, out _);
+        }
+
+        public static object Get(long bid)
+        {
+            return s_registry[bid];
+        }
+    }
+
+    /// <summary>
+    /// Stores the JVMObjectReference object of type org.apache.spark.broadcast.Broadcast for all
+    /// active broadcast variables that are sent to the workers through the CreatePythonFunction.
+    /// This registry is only used on the driver side.
+    /// </summary>
+    internal static class JvmBroadcastRegistry
+    {
+        private static List<JvmObjectReference> s_jvmBroadcastVariables = 
```

Comment:
`readonly`. If your IDE doesn't notify you about this, you may want to check the settings.